internal/utils/common.go

```diff
@@ -135,6 +135,26 @@ func GetTarget(dryRun bool) (string, string, error) {
 	return imgs[0], label, nil
 }
 
+// RebootOrWait will reboot the system or wait forever.
+// It will print the error message before rebooting/waiting
+// If the rd.immucore.rebootonfailure is set in the cmdline, it will reboot.
+// If not, it will wait forever.
+// If the error is not nil it will log it.
+func RebootOrWait(msg string, err error) {
+	if err != nil {
+		Log.Error().Err(err).Msg(msg)
+	}
```

Comment:
It should be safe to call this function with a nil error. It should do nothing in that case.

```suggestion
  if err == nil {
    return
  }
  Log.Error().Err(err).Msg(msg)
```

Reply:
In which case `msg` is not needed. We can simply use the `err.Error()` as message.

Reply:
And the method could be called `RebootOrWaitIfErr` if we go that way.

Reply:
but we want some message becuase this could be used from other parts in which the msg is the context and the error is the error returned by whatever (in this case the unlock)

We want to have context in this to present the error properly no?

And this si a method that its called when we wait to halt/reboot execution. The only thing that the err checking does here is in order to print or not that extra info so it doesnt look ugly. I dont want to print an empty error to make it uglier

Reply:
So currently this will:

 - print an error at Error level if there is any, so we can see the full error (may not be user friendly)
 - print the message at warning level, user friendly
 - tell the user if we are rebooting or halting boot


I think this is the correct approach, we give it info, make the user aware of whats going on and where it failed, and provide an error that its useful for troubleshooting (mainly us)

And this method is specific for calling a reboot/wait not for anything else, so passing a error nil should not affect the outcome of calling this.

Reply:
The caller could prepare an error with the proper message (e.g. by wrapping an previous error) to provide context. I still don't see why we would like to "reboot or wait" if there is no error at all but I don't mind much leaving it as it is. We will see how this is used when we get more cases.

Reply:
for example, we exit when secureboot is not enabled. That has no inherent error from anything and repeating the error would look bad.
The idea would be that this si valid for actual error as well as our own checks (like if sysext are not signed, we want to stop execution with a message saying that X extension is not signed) so it wouold be valid for both usecases easily :D

Reply:
Yes I get it. But you could say that sysext not being signed is indeed an error which we can build before we pass it down to this function. Anyway, doesn't matter much.